src/time/Waveform.cpp

```diff
@@ -31,34 +32,42 @@ void Waveform::store(const Eigen::VectorXd &data)
   this->_timeWindows.col(columnID) = data;
 }
 
-void Waveform::moveToNextWindow(int timeWindows, int order)
+void Waveform::moveToNextWindow(int order)
 {
-  auto initialGuess = extrapolateData(order, timeWindows);
+  if (_numberOfValidSamples < numberOfSamples()) {
+    _numberOfValidSamples++;
+  }
+  auto initialGuess = extrapolateData(order);
```

Comment:
```suggestion
  auto initialGuess = extrapolateData(order);
  if (_numberOfValidSamples < numberOfSamples()) {
    _numberOfValidSamples++;
  }
```
We first have to compute the initial guess and then increase `_numberOfValidSamples`. Otherwise an unitialized sample might be used for the computation of the initial guess.

Reply:
I tried out this buggy implementation on the basis of de659d928083fbde0b96b440ff67e0949264742c and the tests are able to detect this. So there is no need for an additional test.